daemon_test.go

```diff
@@ -112,6 +113,23 @@ func TestDaemonHappyPath(t *testing.T) {
 		require.Equal(t, time.Second*10, ra.msg.RouterLifetime)
 		require.Equal(t, time.Millisecond*10000, ra.msg.ReachableTime)
 		require.Equal(t, time.Millisecond*10000, ra.msg.RetransmitTimer)
+
+		// Assign the value you wish to inspect to mtuValue
+		// Loop through the options included in all RAs to find the MTU option
+		var mtuOption *ndp.MTU
+		const mtuValue = 1500
+
+		for _, option := range ra.msg.Options {
+			if opt, ok := option.(*ndp.MTU); ok {
+				mtuOption = opt
+				break
+			}
+		}
+
+		// If the MTU option is not specified, no inspection by MTU value is performed
+		if mtuOption != nil {
+			require.Equal(t, uint32(mtuValue), mtuOption.MTU)
+		}
```

Comment:
This test passes if there's no MTU option advertised which should be an error.

```suggestion
		// Find MTU option
		var mtuOption *ndp.MTU
		for _, option := range ra.msg.Options {
			if opt, ok := option.(*ndp.MTU); ok {
				mtuOption = opt
				break
			}
		}
		require.NotNil(t, mtuOption, "MTU option is not advertised")
		require.Equal(t, uint32(1500), mtuOption.MTU, "Invalid MTU")
```

Reply:
I fixed